imports/ui/QuestionEditItem.jsx

```diff
@@ -378,31 +378,37 @@ export class QuestionEditItem extends Component {
    * Calls {@link module:questions~"questions.insert" questions.insert} to save question to db
    */
   saveQuestion () {
-    const user = Meteor.user()
-    let question = _.extend({
-      createdAt: new Date(),
-      approved: user.hasGreaterRole('professor') || user.isInstructor(this.props.courseId),
-    }, _.omit(this.state, 'courses'))
-
-    if (question.options.length === 0 && question.type !== QUESTION_TYPE.SA) return
-
-    if (this.props.sessionId) question.sessionId = this.props.sessionId
-    if (this.props.courseId) question.courseId = this.props.courseId
-
-    // insert (or edit)
-    Meteor.call('questions.insert', question, (error, newQuestion) => {
-      if (error) {
-        alertify.error('Error: ' + error.error)
-      } else {
-        if (!this.state._id) {
-          alertify.success('Question Saved')
-          if (this.props.onNewQuestion) this.props.onNewQuestion(newQuestion._id)
-        } else {
-          alertify.success('Edits Saved')
-        }
-        this.setState(newQuestion)
+
+    Meteor.call('courses.getCourseApproved', this.props.courseId, (err, approved) => {
+      if (err) alertify.error('Cannot get question permissions')
+      else {
+        const user = Meteor.user()
+        let question = _.extend({
+          createdAt: new Date(),
+          approved: user.hasGreaterRole('professor') || user.isInstructor(this.props.courseId) || approved,
```

Comment:
I'm not sure this makes sense. The approved property of the question should be set if someone approved the question (or they are an instructor). It should not be a property of the question that the course requires approved questions only.

Reply:
Should setting the require approval property then just set student submitted questions to public? I do think I worded this PR badly as the main change is the ability to let the Professor choose if questions should be approved automatically.

Reply:
I think we have to be clear on the definition of approved. I would think that the approved field in a question is only set to true if a course instructor has approved that question. At the course level, I think we can set a property, as you have, to define whether unapproved questions are allowed -> so it's not that the questions are automatically approved, it's that one does not require the approved field to be true in the publication for the question library and public library. Make sense?

Reply:
Yes I see.